### ISSUE TITLE
Implement schema generator

### ISSUE THREAD

**Issue by peol** [qlik-oss/enigma.js#469]
Based on the new JSON-RPC schema. enigma.js master would update periodically to new engine versions that are bundled with the package, but also offer the generator as part of the package for people that wants to generate an API for a specific engine version.

**Comment by AthmaneQlik** [comment]
Hello,

Do you know if it exist new version of schema (enigma.js) ?
The las version i did found is according to [Qix Engine Version 12.67.2](https://github.com/qlik-oss/enigma.js/tree/master/schemas).
Actual version of Qix Engine is 12.170.2 (Qlik Sense June 2018).

Thank you for your help.
Athmane

**Comment by peol** [comment]
Hi @AthmaneQlik — sorry about that, we've been in vacation mode, just getting back (and catching up). There is a new version of enigma.js published to npm just now: v2.3.0, it should contain `schemas/12.170.2.json`.

**Comment by AthmaneQlik** [comment]
Awsome !!!
thank you for your reactivity and welcome back from your vacation :)

**Comment by provelu-bram** [comment]
Any chance the schema's will be updated to a newer version of the QIX Engine?

**Comment by peol** [comment]
@provelu-bram No plans currently, is there any specific version you're after?

**Comment by provelu-bram** [comment]
Not necessarily, being up-to-date is however a nice to have. If I understand correctly the schemas can be updated using the Qlik Engine JSON API as a reference. If that is the case I can implement the few missing items manualy. The [API insights](https://api-insights.qlik.com/#/api-changes/sense) show the changes nicely so it should not take to much time.
